bot.py

```diff
@@ -0,0 +1,108 @@
+from collections import defaultdict
+from contextlib import contextmanager
+from typing import Text, Dict, List, Generator
+import math
+import requests
+from flask import Flask, request, jsonify
+
+app = Flask(__name__)
+
+inmemory_storage = defaultdict(list)
+
+
+class Conversation(object):
+    def __init__(self, conversation_id: Text, existing_conversation_events: List[Dict]) -> None:
+        """Creates a conversation.
+        Args:
+            existing_conversation_events: Events which happened earlier in this conversation.
+        """
+        self.conversation_id = conversation_id
+        self.existing_conversation_events = existing_conversation_events
+        self.number_existing_conversation_events = len(existing_conversation_events)
+
+    def addd_user_message(self, message: Text) -> None:
+        self.existing_conversation_events.append({"type": "user", "message": message})
+
+    def add_bot_message(self, bot_messages: Text) -> None:
+        self.existing_conversation_events.append({"type": "bot", "message": bot_messages})
+
+    def new_events_dict(self) -> List[Dict]:
+        return self.existing_conversation_events[self.number_existing_conversation_events:]
```

Comment:
Documentation for the methods should be included here too. Also missing the `conversation_id` argument from the current docstring.